### PR TITLE
Date slider and comparison view functionality

### PR DESCRIPTION
**Pull request by teemutontti**
### DateSlider
Has props `value`, `onChange`, `min` and `max`. The component renders a basic slider (one handle) if it gets a single number through the `value`-prop and a range slider (two handles) if it gets an array of two numbers. Slider values are the values that are given from a `dayjs`-objects `valueOf`-method.

NOTE! There is a known bug of the slider drag not working (see issue #69).

### Changes
- Side-by-side view can now be accessed in satellite view by clicking the compare button
- `Controls` now uses position to unify the position of it's children
- `MapComponent` passes multiple states to it's children for handling view changes
- Added `dayjs` as a dependency to handle app's dates

### CODE REVIEW

File: app/components/controls/slider/DateSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import { useMap } from "react-leaflet";
+
+interface DateSliderProps {
+  value: number | number[];
+  onChange: (e: Event, value: number | number[]) => void;
+  min: number;
+  max: number;
+}
+
+// Renders a basic slider (one handle) if value prop holds a single number value
```
Comment (kuukkanen) [inline app/components/controls/slider/DateSlider.tsx:12]: Should this be a doc comment? See https://tsdoc.org/.

**Review by kuukkanen — COMMENTED** [review]
Changing the date from the slider with only one satellite image open (so not in comparison view) doesn't seem to update the image. I have to leave the satellite view and come back in order for the date change to take effect.

File: app/components/controls/slider/DateSlider.tsx (hunk 1 above)
Comment (teemutontti) [reply app/components/controls/slider/DateSlider.tsx:12]: Sure, I'll change it to be a doc comment!

**Comment by teemutontti** [comment]
> Changing the date from the slider with only one satellite image open (so not in comparison view) doesn't seem to update the image. I have to leave the satellite view and come back in order for the date change to take effect.

Hmm🤔 I'll look into this!

**Review by kuukkanen — DISMISSED** [review]

**Review by kuukkanen — CHANGES_REQUESTED** [review]
The history is real messed up now 😅. Don't bring in other peoples changes in your PRs, only resolve merge conflicts with your own commits and don't do merges.

**Comment by teemutontti** [comment]
> Please don't do merges in pull requests!

> The history is real messed up now 😅. Don't bring in other peoples changes in your PRs, only resolve merge conflicts with your own commits and don't do merges.

I don't know why but I had to pull the branch from remote because it was "behind its remote counterpart"? Do you know why this happened and how I should have gone about it?

**Comment by kuukkanen** [comment]
> > Please don't do merges in pull requests!
> 
> > The history is real messed up now 😅. Don't bring in other peoples changes in your PRs, only resolve merge conflicts with your own commits and don't do merges.
> 
> I don't know why but I had to pull the branch from remote because it was "behind its remote counterpart"? Do you know why this happened and how I should've gone about it?

Yes, that's perfectly normal and happens from time to time. The way to fix it is to NOT do merge but rebase instead. E.g. `git rebase main` or similar.

**Comment by kuukkanen** [comment]
Try using `git reflog` and `git reset` to go back to previous state and then do rebase.

**Pull request merged**